File: 4. INTEGRACION NUMERICA/6. Regla del punto medio/6.2 Regla del punto medio compuesta/puntoMedioCompuesto.c
#include <stdio.h>
#include <math.h>

double f(double x){
	/* === DEFINE THE FUCTION TO USE ===*/
	return 0;
}

double puntoMedioCompuesto(double a, double b, int n){
	double h = (b - a)/n;
	double X, XI0 = 0;

	for(int i=1; i<n; i+=2){
		X = a + i*h;
		printf("%lf\n", X);
		XI0 += f(X);
	}

	double XI = 2 * h * XI0;

	printf("\n");
	return XI;
}

int main(){
	/* === INSERT DATA HERE === */
	int n;
	double a, b;

	/* === CALCULATE === */
	double r = puntoMedioCompuesto(a, b, n);
	printf("%lf\n", r);

	/* === EXIT === */
	return 0;
}

File: 4. INTEGRACION NUMERICA/6. Regla del punto medio/6.2 Regla del punto medio compuesta/puntoMedioCompuesto.m
function XI = puntoMedioCompuesto(f, a, b, n)
	h = (b - a)/n;
	X = XI0 = 0;

	for i = 1:2:n-1
		X = a + i*h;
		XI0 += f(X);
	endfor

	XI = 2 * h * XI0;
endfunction
